foundation/plugins/org.jboss.tools.foundation.core/src/org/jboss/tools/foundation/core/properties/internal/SimpleHierarchicalVersion.java

```diff
@@ -24,16 +26,19 @@
 	public SimpleHierarchicalVersion(String version) {
 		internalVersion = version;
 	}
-	
+
+	private static final Pattern MILESTONES_PATTERN = Pattern.compile("^(Alpha|Beta|AM|RC|CR)(\\d+)$");
```

Comment:
There's no Final2 or GA3 to consider. Final/GA versions are already properly processed, and covered by matching test cases.